### PR TITLE
flexibleServerVersion

### PR DESCRIPTION
**Pull request by fatima99s**

### CODE REVIEW

File: plugins/azure/postgresqlserver/flexibleServerVersion.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
+const async = require('async');
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'PostgreSQL Flexible Server Version',
+    category: 'PostgreSQL Server',
+    domain: 'Databases',
+    description: 'Ensure PostgreSQL flexible servers is using the latest server version.',
+    more_info: 'The latest version of PostgreSQL for flexible servers will give access to new software features, resolve reported bugs through security patches, and improve compatibility with other applications and services.',
```
Comment (mehakseedat63) [inline plugins/azure/postgresqlserver/flexibleServerVersion.js:9]:
```suggestion
    more_info: 'Using the latest version of PostgreSQL for flexible servers will give access to new software features, resolve reported bugs through security patches, and improve compatibility with other applications and services.',
```

File: plugins/azure/postgresqlserver/flexibleServerVersion.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    recommended_action: 'Upgrade the version of PostgreSQL flexible server to the latest available version..',
```
Comment (mehakseedat63) [inline plugins/azure/postgresqlserver/flexibleServerVersion.js:10]:
```suggestion
    recommended_action: 'Upgrade the version of PostgreSQL flexible server to the latest available version.',
```

File: plugins/azure/postgresqlserver/flexibleServerVersion.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    link: 'https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/concepts-supported-versions',
+    apis: ['servers:listPostgresFlexibleServer'],   
+    settings: {
+        server_desired_version: {
+            name: 'Postgresql Flexible Server Desired Version',
+            description: 'Desire Postgresql Flexible Server Version ',
+            regex: '^[0-9]+$',
+            default: '11'
+        }
+    },
```
Comment (mehakseedat63) [inline plugins/azure/postgresqlserver/flexibleServerVersion.js:20]:
```suggestion
    settings: {
        server_desired_version: {
            name: 'Postgresql Flexible Server Desired Version',
            description: 'Desire Postgresql Flexible Server Version ',
            regex: '^[0-9]+$',
            default: '11'
        }
    },
```
this shouldn't be a setting, we should find the minimum version which should be used and check against it. This is one of those periodic refactoring plugins where we keep checking for deprecated and outdated versions and update them accordingly (also add it to the periodic refactoring sheet).

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]

File: plugins/azure/postgresqlserver/flexibleServerVersion.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+                helpers.addResult(results, 0, 'No existing PostgreSQL flexible servers found', location);
+                return rcb();
+            }
+
+            for (var flexibleServer of servers.data) {
+                let version = parseFloat(flexibleServer.version);
+
+                if (version && version >= 13) {
+                    helpers.addResult(results, 0,
+                        'Postgresql flexible server has the latest server version', location, flexibleServer.id);
```
Comment (alphadev4) [inline plugins/azure/postgresqlserver/flexibleServerVersion.js:41]:
```suggestion
                        'Postgresql flexible server has the latest postgresql version', location, flexibleServer.id);
```

File: plugins/azure/postgresqlserver/flexibleServerVersion.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+                } else {
+                    helpers.addResult(results, 2,
+                        'Postgresql flexible server does not the latest server version', location, flexibleServer.id);
```
Comment (alphadev4) [inline plugins/azure/postgresqlserver/flexibleServerVersion.js:44]:
```suggestion
                        'PostgreSQL flexible server does not the latest server version', location, flexibleServer.id);
```

File: plugins/azure/postgresqlserver/flexibleServerVersion.spec.js [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+            const cache = createCache([]);
+
+            flexibleServerVersion.run(cache, {}, callback);
+        })
+
+        it('should give failing result if postgresql flexiable server does nothave the latest version', function(done) {
+            const callback = (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(2);
+                expect(results[0].message).to.include('Postgresql flexible server does not the latest server version');
```
Comment (alphadev4) [inline plugins/azure/postgresqlserver/flexibleServerVersion.spec.js:62]:
```suggestion
                expect(results[0].message).to.include('PostgreSQL flexible server does not the latest server version');
```

File: plugins/azure/postgresqlserver/flexibleServerVersion.spec.js [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+            );
+
+            flexibleServerVersion.run(cache, {}, callback);
+        });
+
+        it('should give passing result if postgresql server have the latest version', function(done) {
+            const callback = (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(0);
+                expect(results[0].message).to.include('Postgresql flexible server has the latest server version');
```
Comment (alphadev4) [inline plugins/azure/postgresqlserver/flexibleServerVersion.spec.js:78]:
```suggestion
                expect(results[0].message).to.include('PostgreSQL flexible server has the latest server version');
```

File: plugins/azure/postgresqlserver/flexibleServerVersion.js [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+                return rcb();
+            }
+
+            if (!servers.data.length) {
+                helpers.addResult(results, 0, 'No existing PostgreSQL flexible servers found', location);
+                return rcb();
+            }
+
+            for (var flexibleServer of servers.data) {
+                let version = parseFloat(flexibleServer.version);
```
Comment (alphadev4) [inline plugins/azure/postgresqlserver/flexibleServerVersion.js:37]:
```suggestion
                If (!flexibleServer.id || !flexibleServer.version) return;
                
                let version = parseFloat(flexibleServer.version);
```

**Review by alphadev4 — APPROVED** [review]

File: plugins/azure/postgresqlserver/flexibleServerVersion.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+                return rcb();
+            }
+
+            if (!servers.data.length) {
+                helpers.addResult(results, 0, 'No existing PostgreSQL flexible servers found', location);
+                return rcb();
+            }
+
+            for (var flexibleServer of servers.data) {
+                If (!flexibleServer.id || !flexibleServer.version) return;
```
Comment (alphadev4) [inline plugins/azure/postgresqlserver/flexibleServerVersion.js:37]:
```suggestion
                if(!flexibleServer.id || !flexibleServer.version) return;
```

File: plugins/azure/postgresqlserver/flexibleServerVersion.js [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+                return rcb();
+            }
+
+            if (!servers.data.length) {
+                helpers.addResult(results, 0, 'No existing PostgreSQL flexible servers found', location);
+                return rcb();
+            }
+
+            for (var flexibleServer of servers.data) {
+                if (!flexibleServer.id || !flexibleServer.version) return;
```
Comment (alphadev4) [inline plugins/azure/postgresqlserver/flexibleServerVersion.js:37]:
```suggestion
                if (!flexibleServer.id || !flexibleServer.version) continue;
```

**Review by mehakseedat63 — APPROVED** [review]

**Pull request merged**